MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -43,16 +43,18 @@ static void ms_save_log_buffer_callback(siginfo_t *info, ucontext_t *uap, void *
   }
 
   // Iterate over the buffered logs and write them to disk.
-  for (int i = 0; i < ms_crashes_log_buffer_size; i++) {
-
-    // Make sure not to allocate any memory (e.g. copy).
-    const std::string &data = msCrashesLogBuffer[i].buffer;
-    int fd = open(msCrashesLogBuffer[i].bufferPath.c_str(), O_WRONLY | O_CREAT | O_TRUNC, 0644);
-    if (fd < 0) {
-      return;
+  for (auto it = msCrashesLogBuffer.begin(), end = msCrashesLogBuffer.end(); it != end; ++it) {
+    for (int i = 0; i < ms_crashes_log_buffer_size; i++) {
+      // Make sure not to allocate any memory (e.g. copy).
+      const std::string &data = it->second[i].buffer;
+      const std::string &path = it->second[i].bufferPath;
+      int fd = open(path.c_str(), O_WRONLY | O_CREAT | O_TRUNC, 0644);
+      if (fd < 0) {
+        return;
+      }
```

Comment:
Use `continue` instead of `return` here, since failing to open any one path doesn't imply that all of them will fail.